### PR TITLE
model combination and tests for context feature and spatialRole

### PR DESCRIPTION
**Pull request by Rahgooy**
model combination and tests for context feature and spatialRole

### CODE REVIEW

**Comment by kordjamshidi** [comment]
Sounds great to me. Thanks! @umarmanzoor please review and merge if ok.

File: saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/util/pb.scala [hunk 1]
```diff
@@ -136,9 +140,12 @@ class ProgressBar(_total: Int) extends Output {
     }
     // time left box
     if (showTimeLeft) {
-      val fromStart = (startTime to DateTime.now).millis.toFloat
+      val millis = (startTime to DateTime.now).millis
+      suffix += " elapsed: " + DurationFormatUtils.formatDuration(millis, "HH:mm:ss", true)
+      val fromStart = millis.toFloat
       val left = (fromStart / current) * (total - current)
```
Comment (Rahgooy) [inline saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/util/pb.scala:146]: @kordjamshidi @umarmanzoor 
Added this to have elapsed time as well. Let me know if you don't want this change.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLApp.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+object textApp extends App with Logging {
+
+  import MultiModalSpRLDataModel._
+
+  val isTrain = true
+  val path = if(isTrain) "data/SpRL/2017/clef/train/sprl2017_train.xml" else "data/SpRL/2017/clef/gold/sprl2017_gold.xml"
+  val reader = new NlpXmlReader(path, "SCENE", "SENTENCE", null, null)
+  reader.setIdUsingAnotherProperty("SCENE", "DOCNO")
```
Comment (Rahgooy) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/mSpRL2017/MultiModalSpRLApp.scala:56]: this, adds `id` to scenes using `DOCNO`.

File: saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/Dictionaries.scala [hunk 3]
```diff
@@ -17,8 +17,19 @@ object Dictionaries {
     "behind", "beneath", "beside", "between", "by", "down", "during", "for", "from", "in", "inside",
     "into", "like", "of", "off", "on", "onto", "over", "round", "through", "to", "towards", "with"
   )
+  val spatial = HashSet(
+    "about", "above", "across", "after", "against", "along", "among", "around", "as", "at", "before",
+    "behind", "beneath", "beside", "between", "by", "down", "during", "for", "from", "in", "inside",
+    "into", "like", "of", "off", "on", "onto", "over", "round", "through", "to", "towards", "with"
+  )
+
```
Comment (kordjamshidi) [inline saul-examples/src/main/scala/edu/illinois/cs/cogcomp/saulexamples/nlp/SpatialRoleLabeling/Dictionaries.scala:25]: so here we needed to add "left" and "right"

**Comment by kordjamshidi** [comment]
We need to update the spatial lex but even now, this is good to merge for me.

**Pull request merged**